streams/src/main/java/org/apache/kafka/streams/ApiUtils.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+
+/**
+ */
+public final class ApiUtils {
```

Comment:
I am wondering, what we gain by introducing this class?

Reply:
Hello @mjsax 

As discussed in [KIP thread](https://www.mail-archive.com/search?l=dev@kafka.apache.org&q=subject:%22%5C%5BDISCUSS%5C%5D+KIP%5C-358%5C%3A+Migrate+Streams+API+to+Duration+instead+of+long+ms+times%22&o=newest&f=1) we want to wrap NPE and `ArithmeticException` into `IllegalArgumentException`

See @vvcephei suggestion: "I still feel that surfacing the ArithmeticException directly would not be a great experience, so I still advocate for wrapping it in an IllegalArgumentException that explains our upper bound for Duration is "max-long number of milliseconds"